### PR TITLE
boards: Update Xiao_BLE Sense DTS for on board pdm microphone

### PR DESCRIPTION
**Pull request by zakport**
Create regulator on GPIO for microphone supply
Set pdm data and clk pins in pinctrl
Add xiao_ble_nrf54840_sense.overlay for dmic sample support 
Change the default mono channel in the sample from left to  right to work with 
the hardware configuration of the Xiao

### CODE REVIEW

File: boards/seeed/xiao_ble/xiao_ble_nrf52840_sense.dts [hunk 1]
```diff
@@ -13,6 +13,13 @@
 	model = "Seeed XIAO BLE Sense";
 	compatible = "seeed,xiao-ble", "seeed,xiao-ble-sense";
 
+	msm261d3526hicpm-c-en {
+		compatible = "regulator-fixed";
+		enable-gpios = <&gpio1 10 (NRF_GPIO_DRIVE_S0H1 | GPIO_ACTIVE_HIGH)>;
+		regulator-name = "MSM261D3526HICPM-C-EN";
+		regulator-boot-on;
```
Comment (kartben) [inline boards/seeed/xiao_ble/xiao_ble_nrf52840_sense.dts:20]: i would move this property to the sample overlay, as I don't think users of this board would want the MIC to be always on by default

File: samples/drivers/audio/dmic/src/main.c [hunk 2]
```diff
@@ -108,7 +108,7 @@ int main(void)
 
 	cfg.channel.req_num_chan = 1;
 	cfg.channel.req_chan_map_lo =
-		dmic_build_channel_map(0, 0, PDM_CHAN_LEFT);
+		dmic_build_channel_map(0, 0, PDM_CHAN_RIGHT);
```
Comment (kartben) [inline samples/drivers/audio/dmic/src/main.c:111]: this change seems unrelated to the XIAO BLE Sense work, so at a minimum it should be a different commit. It looks like a more robust change would be to do LEFT, then RIGHT, then LEFT+RIGHT?

File: boards/seeed/xiao_ble/xiao_ble_nrf52840_sense.dts [hunk 3]
```diff
@@ -38,3 +45,11 @@
 		status = "okay";
 	};
 };
+
+&pdm0 {
+	status = "okay";
```
Comment (kartben) [inline boards/seeed/xiao_ble/xiao_ble_nrf52840_sense.dts:50]: same, would be better to "okay" this only in the overlay.

**Review by kartben — CHANGES_REQUESTED** [review]

File: samples/drivers/audio/dmic/src/main.c (hunk 2 above)
Comment (zakport) [reply samples/drivers/audio/dmic/src/main.c]: Yeah I agree.

**Review by kartben — DISMISSED** [review]

File: samples/drivers/audio/dmic/boards/xiao_ble_nrf52840_sense.overlay [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+ */
+/ {
+	msm261d3526hicpm-c-en {
+		regulator-boot-on;
+	};
+};
+
+dmic_dev: &pdm0 {
+  status = "okay";
+};
```
Comment (mniestroj) [inline samples/drivers/audio/dmic/boards/xiao_ble_nrf52840_sense.overlay:12]:
```suggestion
dmic_dev: &pdm0 {
	status = "okay";
};
```
Please replace "double space" with TAB.

**Review by mniestroj — COMMENTED** [review]
Changes LGTM, just a small whitespace suggestion.

**Review by kartben — APPROVED** [review]

**Review by mniestroj — APPROVED** [review]
Thanks!
